data/actionsmap/zsh_completion.py

```diff
@@ -0,0 +1,665 @@
+#!/usr/bin/python3
+import re
+import sys
+import yaml
+from pathlib import Path
+
+# DOCS:
+# - https://github.com/zsh-users/zsh/blob/master/Etc/completion-style-guide
+# - http://zsh.sourceforge.net/Doc/Release/Completion-System.html#Completion-System
+#   or `man zshcompsys`
+# - http://zsh.sourceforge.net/Guide/zshguide06.html
+#
+# Misc:
+# - http://zsh.sourceforge.net/Doc/Release/Parameters.html#Array-Parameters
+
+# TODO:
+# - use the extra:required:True pattern (similar to `nargs`?)
+# - Have the global options listed in a separate category
+# - Allow multiple arguments per option
+#   (e.g.: `yunohost user info alice --fields uid cn mail`)
+# - In `yunohost.yml`, consider merging:
+#   - metavar
+#   - pattern
+#   - autocomplete
+# - Make use of `type`, maybe using `_guard`
+# - Use `pattern`, maybe with `_guard`. This seems hard though, as ZSH has
+# its own globbing language...
+# Link about this globbing system:
+# http://zsh.sourceforge.net/Doc/Release/Expansion.html#Filename-Generation
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# - Optimization:
+#   - caching mecanism: invalidate the cache afer some commands? Hard, the
+#   cache is local to user
+#   - implement a zstyle switch, to change the cache validity period?
+
+ZSH_COMPLETION_FILE = '/usr/local/share/zsh/site-functions/_yunohost'
```

Comment:
Hey there !

Small issue to fix for the debian packaging : that file won't be packaged. Instead, if we stick to something similar what's done for the bash completion : the output should go to some folder inside the yunohost/ tree
([for example bash completion goes into bash-completion.d/yunohost](https://github.com/YunoHost/yunohost/blob/dev/data/actionsmap/yunohost_completion.py#L16)) then gotta add [a rule inside debian/install to tell it the source/destination during the actual install of the .deb](https://github.com/YunoHost/yunohost/blob/dev/debian/install#L3)

Reply:
Thanks! I didn't see this Debian packaging aspect. I just pushed something in this regard.